app/scripts/modules/core/src/pipeline/config/stages/entityTags/ApplyEntityTagsStageConfig.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import Select, { Option } from 'react-select';
+
+import { IEntityRef, IEntityTag, IStage } from 'core/domain';
+import { IStageConfigProps, StageConfigField } from 'core/pipeline';
+import { SETTINGS } from 'core/config/settings';
+
+import { TagEditor } from './TagEditor';
+import { FormField, ReactSelectInput, StringsAsOptions } from 'core/presentation';
+
+export interface IApplyEntityTagsStage extends IStage {
+  entityRef: IEntityRef;
+  tags: IEntityTag[];
+}
+
+export interface IApplyEntityTagsStageConfigProps extends IStageConfigProps {
+  stage: IApplyEntityTagsStage;
+}
+
+export class ApplyEntityTagsStageConfig extends React.Component<IApplyEntityTagsStageConfigProps> {
+  private entityOptions = [
+    { label: 'Server Group', value: 'servergroup' },
+    { label: 'Load Balancer', value: 'loadbalancer' },
+    { label: 'Application', value: 'application' },
+    { label: 'Cluster', value: 'cluster' },
+    { label: 'Security Group', value: 'securitygroup' },
+  ];
+
+  private entityRefTypeChanged = (entityType: string) => {
+    const entityRef = { entityType };
+    this.props.updateStageField({ entityRef });
+  };
+
+  private entityRefFieldChanged = (key: string, value: string) => {
+    const entityRef = { ...this.props.stage.entityRef, ...{ [key]: value } };
+    this.props.updateStageField({ entityRef });
+  };
+
+  private tagChanged = (tag: IEntityTag, index: number) => {
+    const tags = this.props.stage.tags.slice();
+    tags[index] = tag;
+    this.props.updateStageField({ tags });
+  };
+
+  private addTag = () => {
+    this.props.stage.tags.push({ name: '', value: '' });
+    this.props.updateStageField({ tags: this.props.stage.tags });
+  };
+
+  public render() {
+    const {
+      stage: { entityRef = {} as IEntityRef, tags = [] as IEntityTag[] },
+      application: {
+        attributes: { cloudProviders = SETTINGS.defaultProviders },
+      },
+    } = this.props;
+
+    return (
+      <div className="form-horizontal">
+        <StageConfigField label="Entity Type">
+          <Select
+            clearable={false}
+            options={this.entityOptions}
+            onChange={(o: Option<string>) => this.entityRefTypeChanged(o.value)}
+            value={entityRef.entityType || ''}
+          />
+        </StageConfigField>
+        {entityRef.entityType && (
+          <>
+            <StageConfigField label="Name">
+              <input
+                className="form-control input-sm"
+                value={entityRef.entityId || ''}
+                onChange={e => this.entityRefFieldChanged('entityId', e.target.value)}
+              />
+            </StageConfigField>
+            {entityRef.entityType !== 'application' && (
+              <>
+                <StageConfigField label="Cloud Provider">
+                  <FormField
```

Comment:
curious why you used `FormField` here but nowhere else?

Reply:
I only used it to use ReactSelectInput. I guess it makes sense to use it for "all the things"?